crates/viewer/re_view_spatial/tests/transform_clamping.rs

```diff
@@ -0,0 +1,261 @@
+                    (2.0, 2.0, 2.0),
+                ])
+                .with_colors([0xFF0000FF]),
+            )
+        });
+
+        test_context.log_entity("spheres/ignored_colors".into(), |builder| {
+            builder.with_archetype(
+                RowId::new(),
+                TimePoint::default(),
+                &re_types::archetypes::Ellipsoids3D::from_centers_and_half_sizes(
+                    [(5.0, 0.0, 0.0)],
+                    [(1.0, 1.0, 1.0)],
+                )
+                .with_colors([0x00FF00FF, 0xFF00FFFF]),
+            )
+        });
+
+        test_context.log_entity("spheres/more_transforms_than_sizes".into(), |builder| {
+            builder
+                .with_archetype(
+                    RowId::new(),
+                    TimePoint::default(),
+                    &re_types::archetypes::Ellipsoids3D::from_centers_and_half_sizes(
+                        [(0.0, 5.0, 0.0)],
+                        [(1.0, 1.0, 1.0)],
+                    )
+                    .with_colors([0x0000FFFF]),
+                )
+                .with_archetype(
+                    RowId::new(),
+                    TimePoint::default(),
+                    &re_types::archetypes::InstancePoses3D::new()
+                        .with_scales([(1.0, 1.0, 1.0), (2.0, 2.0, 2.0)]),
+                )
+        });
+
+        test_context.log_entity("spheres/no_primaries".into(), |builder| {
+            builder
+                .with_archetype(
+                    RowId::new(),
+                    TimePoint::default(),
+                    &re_types::archetypes::Ellipsoids3D::update_fields()
+                        .with_centers([(5.0, 0.0, 0.0)])
+                        .with_colors([0xFF00FFFF]),
+                )
+                .with_archetype(
+                    RowId::new(),
+                    TimePoint::default(),
+                    &re_types::archetypes::InstancePoses3D::new()
+                        .with_scales([(1.0, 1.0, 1.0), (2.0, 2.0, 2.0)]),
+                )
+        });
+    }
+
+    let view_ids = setup_blueprint(&mut test_context);
+    run_view_ui_and_save_snapshot(
+        &mut test_context,
+        view_ids,
+        "transform_clamping",
+        egui::vec2(300.0, 300.0),
+    );
+}
+
+fn get_test_context() -> TestContext {
+    let mut test_context = TestContext::default();
+
+    // It's important to first register the view class before adding any entities,
+    // otherwise the `VisualizerEntitySubscriber` for our visualizers doesn't exist yet,
+    // and thus will not find anything applicable to the visualizer.
+    test_context.register_view_class::<re_view_spatial::SpatialView3D>();
+
+    // Make sure we can draw stuff in the hover tables.
+    test_context.component_ui_registry = re_component_ui::create_component_ui_registry();
+    // Also register the legacy UIs.
+    re_data_ui::register_component_uis(&mut test_context.component_ui_registry);
```

Comment:
better suited for a follow-up: can we simply move that to the harness
(also not really needed for this test?)

Reply:
Yeah i'll do a follow-up tomorrow